uplc-hash.md

```diff
@@ -0,0 +1,18 @@
+# UPLC Hash
+
+```k
+requires "domains.md"
+requires "uplc-syntax.md"
+requires "krypto.md"
+
+module UPLC-HASH
+  imports INT
+  imports STRING
+  imports KRYPTO
+  imports K-REFLECTION
+  imports UPLC-SYNTAX
+
+  syntax Int ::= #uplcHash(Value) [function]
+  rule #uplcHash(V:Value) => String2Base(Sha3_256(#unparseKORE(V)), 16)
```

Comment:
This may be a performance bottleneck in the future. I would put a comment here at least that we should look for a faster hashing scheme, because serializing to Kore is an extra process call, and the calling sha3_256 also takes time. The LLVM backend already natively is computing hashes of things, and can do that directly if we can expose that functionality.

Reply:
Done